### PR TITLE
[DONE] Translating "Adopting Concurrent Mode"

### PR DESCRIPTION
**Pull request by mohamedsgap**
Started translating "Adopting Concurrent Mode"

### CODE REVIEW

**Comment by mohamedsgap** [comment]
Hi guys, 
@3imed-jaberi, @Aissaoui-Ahmed, @aladin002dz, @asantarissy and @Fcmam5
just finished translated 'Adopting Concurrent Mode' I need your review :)

File: content/docs/concurrent-mode-adoption.md [hunk 1]
```diff
@@ -31,36 +31,37 @@ next: concurrent-mode-reference.html
...
+يهدف هذا الإصدار في المقام الأول إلى المتبنين في وقت مبكر، ومؤلفي المكتبات ، والأشخاص الفضوليين.
 
-### Enabling Concurrent Mode {#enabling-concurrent-mode}
+نحن نستخدم هذه الشفرة في الإنتاج (وهو مناسب لنا) ولكن لا تزال هناك بعض الأخطاء والميزات المفقودة والفجوات في الوثائق. نود أن نسمع المزيد حول ما ينقطع في الوضع المتزامن حتى نتمكن من إعداده بشكل أفضل لإصدار رسمي مستقر في المستقبل.
 
-Normally, when we add features to React, you can start using them immediately. Fragments, Context, and even Hooks are examples of such features. You can use in new code without making any changes to the existing code.
+### تمكين الوضع المتزامن {#enabling-concurrent-mode}
 
-Concurrent Mode is different. It introduces semantic changes to how React works. Otherwise, the [new features](/docs/concurrent-mode-patterns.html) enabled by it *wouldn't be possible*. This is why they're grouped into a new "mode" rather than released one by one in isolation.
+عادة، عندما نضيف ميزات إلى React، يمكنك البدء في استخدامها على الفور. الشظايا والسياق وحتى الخطافات هي أمثلة على هذه الميزات. يمكنك استخدام الكود الجديد دون إجراء أي تغييرات على الكود الموجود.
```
Comment (Aissaoui-Ahmed) [inline content/docs/concurrent-mode-adoption.md]:
```suggestion
عادة، عندما نضيف ميزات إلى React ،يمكنك البدء في استخدامها على الفور. الأجزاء والسياق وحتى الخطافات هي أمثلة على هذه الميزات. يمكنك استخدام الكود الجديد دون إجراء أي تغييرات على الكود الموجود.
```

File: content/docs/concurrent-mode-adoption.md [hunk 2]
```diff
@@ -136,6 +138,6 @@ You can think of the Blocking Mode as a "gracefully degraded" version of the Con
 
 </div>
 
-\*: Legacy mode has automatic batching in React-managed events but it's limited to one browser task. Non-React events must opt-in using `unstable_batchedUpdates`. In Blocking Mode and Concurrent Mode, all `setState`s are batched by default.
+\*: يحتوي الوضع القديم على مجموعة تلقائية في الأحداث التي تتم إدارتها بواسطة React، ولكنه يقتصر على مهمة مستعرض واحدة. يجب أن تشترك الأحداث التي لا تتضمن رد الفعل في استخدام `unstable_batchedUpdates`. في وضع الحظر و الوضع المتزامن، يتم تجميع جميع`setState`s  افتراضيًا.
```
Comment (Aissaoui-Ahmed) [inline content/docs/concurrent-mode-adoption.md]:
```suggestion
\*: يحتوي الوضع القديم على مجموعة تلقائية في الأحداث التي تتم إدارتها بواسطة React، ولكنه يقتصر على مهمة مستعرض واحدة. يجب أن تشترك الأحداث التي لا تتضمن رد الفعل في استخدام `unstable_batchedUpdates`. في وضع الحظر و الوضع المتزامن، يتم تجميع جميع`setState` افتراضيًا.
```

File: content/docs/concurrent-mode-adoption.md [hunk 3]
```diff
@@ -76,37 +77,38 @@ ReactDOM.createRoot(
...
 
-To recap:
+لكى نلخص:
 
-* **Legacy Mode:** `ReactDOM.render(<App />, rootNode)`. This is what React apps use today. There are no plans to remove the legacy mode in the observable future — but it won't be able to support these new features.
-* **Blocking Mode:** `ReactDOM.createBlockingRoot(rootNode).render(<App />)`. It is currently experimental. It is intended as a first migration step for apps that want to get a subset of Concurrent Mode features.
-* **Concurrent Mode:** `ReactDOM.createRoot(rootNode).render(<App />)`. It is currently experimental. In the future, after it stabilizes, we intend to make it the default React mode. This mode enables *all* the new features.
+* **الوضع القديم:** `ReactDOM.render (<App/> ، rootNode)`. هذا هو ما تستخدمه تطبيقات React اليوم. لا توجد خطط لإزالة الوضع القديم في المستقبل الذي يمكن ملاحظته - لكنه لن يكون قادرًا على دعم هذه الميزات الجديدة.
```
Comment (Aissaoui-Ahmed) [inline content/docs/concurrent-mode-adoption.md]:
```suggestion
* **الوضع القديم:** `ReactDOM.render (<App /> ، rootNode)`. هذا هو ما تستخدمه تطبيقات React اليوم. لا توجد خطط لإزالة الوضع القديم في المستقبل الذي يمكن ملاحظته - لكنه لن يكون قادرًا على دعم هذه الميزات الجديدة.
```

File: content/docs/concurrent-mode-adoption.md [hunk 4]
```diff
@@ -76,37 +77,38 @@ ReactDOM.createRoot(
...
-In our experience, code that uses idiomatic React patterns and doesn't rely on external state management solutions is the easiest to get running in the Concurrent Mode. We will describe common problems we've seen and the solutions to them separately in the coming weeks.
+### خطوة الترحيل: وضع الحظر {#migration-step-blocking-mode}
 
-### Migration Step: Blocking Mode {#migration-step-blocking-mode}
+بالنسبة لقواعد الشفرات القديمة، قد يكون الوضع المتزامن خطوة بعيدة جدًا. هذا هو السبب في أننا نقدم أيضًا "وضع الحظر" الجديد في تصميمات React التجريبية. يمكنك تجربتها عن طريق استبدال `createRoot` بـ` createBlockingRoot`. إنه يقدم مجموعة *صغيرة* فقط من ميزات الوضع المتزامن، ولكنه أقرب إلى كيفية عمل React اليوم ويمكن أن يكون بمثابة خطوة ترحيل.
 
-For older codebases, Concurrent Mode might be a step too far. This is why we also provide a new "Blocking Mode" in the experimental React builds. You can try it by substituting `createRoot` with `createBlockingRoot`. It only offers a *small subset* of the Concurrent Mode features, but it is closer to how React works today and can serve as a migration step.
 
-To recap:
+لكى نلخص:
```
Comment (Aissaoui-Ahmed) [inline content/docs/concurrent-mode-adoption.md]:
```suggestion
لكي نلخص:
```

File: content/docs/concurrent-mode-adoption.md [hunk 5]
```diff
@@ -31,36 +31,37 @@ next: concurrent-mode-reference.html
...
+### تمكين الوضع المتزامن {#enabling-concurrent-mode}
 
-Concurrent Mode is different. It introduces semantic changes to how React works. Otherwise, the [new features](/docs/concurrent-mode-patterns.html) enabled by it *wouldn't be possible*. This is why they're grouped into a new "mode" rather than released one by one in isolation.
+عادة، عندما نضيف ميزات إلى React، يمكنك البدء في استخدامها على الفور. الشظايا والسياق وحتى الخطافات هي أمثلة على هذه الميزات. يمكنك استخدام الكود الجديد دون إجراء أي تغييرات على الكود الموجود.
 
-You can't opt into Concurrent Mode on a per-subtree basis. Instead, to opt in, you have to do it in the place where today you call `ReactDOM.render()`.
+الوضع المتزامن مختلف. أنه يقدم تغييرات الدلالية على كيفية عمل React. خلاف ذلك، فإن [الميزات الجديدة](/docs/concurrent-mode-patterns.html) الممكَّنة به *لن تكون ممكنة*. لهذا السبب تم تجميعهم في "وضع" جديد بدلاً من إصدار واحد تلو الآخر في عزلة.
 
-**This will enable Concurrent Mode for the whole `<App />` tree:**
+لا يمكنك الاشتراك في الوضع المتزامن على أساس الشجرة الفرعية. بدلاً من ذلك، للاشتراك، يتعين عليك القيام بذلك في المكان الذي تقوم بنداء فيه اليوم بـ `ReactDOM.render () `.
```
Comment (Aissaoui-Ahmed) [inline content/docs/concurrent-mode-adoption.md]:
```suggestion
لا يمكنك الاشتراك في الوضع المتزامن على أساس الشجرة الفرعية. بدلاً من ذلك، للاشتراك، يتعين عليك القيام بذلك في المكان الذي تقوم بنداء فيه اليوم بـ `ReactDOM.render()`.
```

File: content/docs/concurrent-mode-adoption.md [hunk 6]
```diff
@@ -76,37 +77,38 @@ ReactDOM.createRoot(
...
+
+في الوضع المتزامن، تعد أساليب دورة الحياة [التي تم تمييزها مسبقًا](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html) باعتبارها "غير آمنة" في الواقع *تكون* غير آمنة، وتؤدي إلى الخلل أكثر مما كانت عليه في رد فعل اليوم. لا نوصي بتجربة وضع المتزامنة حتى يكون تطبيق [Strict Mode](https://reactjs.org/docs/strict-mode.html) متوافقًا.
 
-In Concurrent Mode, the lifecycle methods [previously marked](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html) as "unsafe" actually *are* unsafe, and lead to bugs even more than in today's React. We don't recommend trying Concurrent Mode until your app is [Strict Mode](https://reactjs.org/docs/strict-mode.html)-compatible.
+## ماذا تتوقع {#what-to-expect}
 
-## What to Expect {#what-to-expect}
+إذا كان لديك تطبيق كبير موجود، أو إذا كان تطبيقك يعتمد على الكثير من حزم الطرف الثالث ، فالرجاء عدم توقع أنه يمكنك استخدام الوضع المتزامن على الفور. **على سبيل المثال، في Facebook، نستخدم "الوضع المتزامن" لموقع الويب الجديد ، لكننا لا نخطط لتمكينه على الموقع القديم.** وذلك لأن موقعنا القديم لا يزال يستخدم أساليب دورة حياة غير آمنة في رمز المنتج، غير متوافق مكتبات الطرف الثالث، والأنماط التي لا تعمل بشكل جيد مع وضع المتزامنة.
```
Comment (3imed-jaberi) [inline content/docs/concurrent-mode-adoption.md]:
```suggestion
إذا كان لديك تطبيق كبير موجود، أو إذا كان تطبيقك يعتمد على الكثير من حزم الطرف الثالث ، فالرجاء عدم توقع أنه يمكنك استخدام الوضع المتزامن على الفور. **على سبيل المثال، في Facebook، نستخدم "الوضع المتزامن" لموقع الويب الجديد ، لكننا لا نخطط لتمكينه على الموقع القديم.** وذلك لأن موقعنا القديم لا يزال يستخدم أساليب دورة حياة غير آمنة في شيفرة الإنتاج، مكتبات طرف ثالث غير متوافق، والأنماط التي لا تعمل بشكل جيد مع الوضع المتزامنة.
```

File: content/docs/concurrent-mode-adoption.md [hunk 7]
```diff
@@ -76,37 +77,38 @@ ReactDOM.createRoot(
...
 
-If you have a large existing app, or if your app depends on a lot of third-party packages, please don't expect that you can use the Concurrent Mode immediately. **For example, at Facebook we are using Concurrent Mode for the new website, but we're not planning to enable it on the old website.** This is because our old website still uses unsafe lifecycle methods in the product code, incompatible third-party libraries, and patterns that don't work well with the Concurrent Mode.
+من خلال تجربتنا، فإن الشفرة التى تستخدم أنماط رد الفعل الاصطلاحية ولا يعتمد على حلول إدارة الحالة الخارجية هو الأسهل لتشغيله في الوضع المتزامن. سنصف المشكلات الشائعة التي رأيناها والحلول الخاصة بها بشكل منفصل في الأسابيع المقبلة.
```
Comment (3imed-jaberi) [inline content/docs/concurrent-mode-adoption.md]: 😠 ...
```suggestion
من خلال تجربتنا، فإن الشفرة التى تستخدم أنماط React الاصطلاحية ولا تعتمد على حلول إدارة الحالة الخارجية هو الأسهل لتشغيله في الوضع المتزامن. سنصف المشكلات الشائعة التي رأيناها والحلول الخاصة بها بشكل منفصل في الأسابيع المقبلة.
```

File: content/docs/concurrent-mode-adoption.md [hunk 8]
```diff
@@ -76,37 +77,38 @@ ReactDOM.createRoot(
...
 
-In our experience, code that uses idiomatic React patterns and doesn't rely on external state management solutions is the easiest to get running in the Concurrent Mode. We will describe common problems we've seen and the solutions to them separately in the coming weeks.
+### خطوة الترحيل: وضع الحظر {#migration-step-blocking-mode}
 
-### Migration Step: Blocking Mode {#migration-step-blocking-mode}
+بالنسبة لقواعد الشفرات القديمة، قد يكون الوضع المتزامن خطوة بعيدة جدًا. هذا هو السبب في أننا نقدم أيضًا "وضع الحظر" الجديد في تصميمات React التجريبية. يمكنك تجربتها عن طريق استبدال `createRoot` بـ` createBlockingRoot`. إنه يقدم مجموعة *صغيرة* فقط من ميزات الوضع المتزامن، ولكنه أقرب إلى كيفية عمل React اليوم ويمكن أن يكون بمثابة خطوة ترحيل.
```
Comment (3imed-jaberi) [inline content/docs/concurrent-mode-adoption.md]:
```suggestion
بالنسبة لقواعد الشفرات القديمة، قد يكون الوضع المتزامن خطوة بعيدة جدًا. هذا هو السبب في أننا نقدم أيضًا "وضع الحظر" الجديد في بناءات React التجريبية. يمكنك تجربتها عن طريق استبدال `createRoot` بـ` createBlockingRoot`. إنه يقدم مجموعة *صغيرة* فقط من ميزات الوضع المتزامن، ولكنه أقرب إلى كيفية عمل React اليوم ويمكن أن يكون بمثابة خطوة ترحيل.
```

File: content/docs/concurrent-mode-adoption.md [hunk 9]
```diff
@@ -76,37 +77,38 @@ ReactDOM.createRoot(
...
 
-To recap:
+لكي نلخص:
 
-* **Legacy Mode:** `ReactDOM.render(<App />, rootNode)`. This is what React apps use today. There are no plans to remove the legacy mode in the observable future — but it won't be able to support these new features.
-* **Blocking Mode:** `ReactDOM.createBlockingRoot(rootNode).render(<App />)`. It is currently experimental. It is intended as a first migration step for apps that want to get a subset of Concurrent Mode features.
-* **Concurrent Mode:** `ReactDOM.createRoot(rootNode).render(<App />)`. It is currently experimental. In the future, after it stabilizes, we intend to make it the default React mode. This mode enables *all* the new features.
+* **الوضع القديم:** `ReactDOM.render (<App /> ، rootNode)`. هذا هو ما تستخدمه تطبيقات React اليوم. لا توجد خطط لإزالة الوضع القديم في المستقبل الذي يمكن ملاحظته - لكنه لن يكون قادرًا على دعم هذه الميزات الجديدة.
+* **وضع الحظر:** `ReactDOM.createBlockingRoot(rootNode).render(<App/>)`. انها حاليا التجريبية. الغرض منه هو خطوة الترحيل الأولى للتطبيقات التي ترغب في الحصول على مجموعة فرعية من ميزات الوضع المتزامن.
```
Comment (3imed-jaberi) [inline content/docs/concurrent-mode-adoption.md]:
```suggestion
* **وضع الحظر:** `ReactDOM.createBlockingRoot(rootNode).render(<App/>)`. حاليا تحت التجربة. الغرض منه هو خطوة الترحيل الأولى للتطبيقات التي ترغب في الحصول على مجموعة فرعية من ميزات الوضع المتزامن.
```

File: content/docs/concurrent-mode-adoption.md [hunk 10]
```diff
@@ -76,37 +77,38 @@ ReactDOM.createRoot(
...
+* **الوضع المتزامن:** `ReactDOM.createRoot(rootNode).render(<App/>)`. انها حاليا التجريبية. في المستقبل، بعد استقراره، نعتزم جعله وضع React الافتراضي. يتيح هذا الوضع *جميع* الميزات الجديدة.
```
Comment (3imed-jaberi) [inline content/docs/concurrent-mode-adoption.md]:
```suggestion
* **الوضع المتزامن:** `ReactDOM.createRoot(rootNode).render(<App/>)`. حاليا تحت التجربة. في المستقبل، بعد استقراره، نعتزم جعله وضع React الافتراضي. يتيح هذا الوضع *جميع* الميزات الجديدة.
```

File: content/docs/concurrent-mode-adoption.md [hunk 11]
```diff
@@ -76,37 +77,38 @@ ReactDOM.createRoot(
...
+من الناحية العملية، نتوقع أن تتمكن معظم التطبيقات التي تستخدم الوضع القديم من الانتقال إلى وضع الحظر على الأقل (إن لم يكن الوضع المتزامن). يمكن أن يكون هذا التجزئة مزعجًا بالنسبة للمكتبات التي تهدف إلى دعم جميع الأوضاع على المدى القصير. ومع ذلك، فإن نقل النظام الإيكولوجي تدريجياً بعيدًا عن Legacy Mode سيؤدي أيضًا إلى *حل* المشكلات التي تؤثر على المكتبات الرئيسية في نظام React، مثل [مربكة سلوك Suspense عند قراءة التخطيط](https://github.com/facebook/react/issues / 14536) و [عدم وجود ضمانات مجمعة متسقة](https://github.com/facebook/react/issues/15080). هناك عدد من الأخطاء التي لا يمكن إصلاحها في الوضع القديم دون تغيير الدلالات، ولكن لا توجد في أوضاع الحظر والتزامن.
```
Comment (3imed-jaberi) [inline content/docs/concurrent-mode-adoption.md]: please @asantarissy , @Fcmam5 , @Aissaoui-Ahmed review this line + 109 ..

File: content/docs/concurrent-mode-adoption.md [hunk 12]
```diff
@@ -116,7 +118,7 @@ You can think of the Blocking Mode as a "gracefully degraded" version of the Con
 
 <div id="feature-table">
 
-|   |Legacy Mode  |Blocking Mode  |Concurrent Mode  |
+|   |الوضع القديم  |وضع الحظر |الوضع المتزامن |
```
Comment (3imed-jaberi) [inline content/docs/concurrent-mode-adoption.md]:
```suggestion
|   |الوضع القديم |وضع الحظر  |الوضع المتزامن |
```

File: content/docs/concurrent-mode-adoption.md [hunk 13]
```diff
@@ -136,6 +138,6 @@ You can think of the Blocking Mode as a "gracefully degraded" version of the Con
 
 </div>
 
-\*: Legacy mode has automatic batching in React-managed events but it's limited to one browser task. Non-React events must opt-in using `unstable_batchedUpdates`. In Blocking Mode and Concurrent Mode, all `setState`s are batched by default.
+\*: يحتوي الوضع القديم على مجموعة تلقائية في الأحداث التي تتم إدارتها بواسطة React، ولكنه يقتصر على مهمة مستعرض واحدة. يجب أن تشترك الأحداث التي لا تتضمن رد الفعل في استخدام `unstable_batchedUpdates`. في وضع الحظر و الوضع المتزامن، يتم تجميع جميع`setState` افتراضيًا.
 
-\*\*: Warns in development.
+\*\*: تحذير فى عملية تطوير.
```
Comment (3imed-jaberi) [inline content/docs/concurrent-mode-adoption.md]:
```suggestion
\*\*: تحذيرات فى عملية تطوير.
```

File: content/docs/concurrent-mode-adoption.md [hunk 14]
```diff
@@ -136,6 +138,6 @@ You can think of the Blocking Mode as a "gracefully degraded" version of the Con
 
 </div>
 
-\*: Legacy mode has automatic batching in React-managed events but it's limited to one browser task. Non-React events must opt-in using `unstable_batchedUpdates`. In Blocking Mode and Concurrent Mode, all `setState`s are batched by default.
+\*: يحتوي الوضع القديم على مجموعة تلقائية في الأحداث التي تتم إدارتها بواسطة React، ولكنه يقتصر على مهمة مستعرض واحدة. يجب أن تشترك الأحداث التي لا تتضمن رد الفعل في استخدام `unstable_batchedUpdates`. في وضع الحظر و الوضع المتزامن، يتم تجميع جميع`setState` افتراضيًا.
```
Comment (3imed-jaberi) [inline content/docs/concurrent-mode-adoption.md]: 😠 ..
```suggestion
\*: يحتوي الوضع القديم على مجموعة تلقائية في الأحداث التي تتم إدارتها بواسطة React، ولكنه يقتصر على مهمة المتصفح الواحيد. يجب أن تشترك الأحداث التي لا تتضمن React في استخدام `unstable_batchedUpdates`. في وضع الحظر و الوضع المتزامن، يتم تجميع جميع `setState` افتراضيًا.
```

**Pull request merged**